### DISCUSSION TITLE
Dependency downloads

### DISCUSSION THREAD

**Discussion by rnicoll** [dogecoin/dogecoin#2256] [Help needed]
If someone wants to jump into development, https://github.com/dogecoin/dogecoin/issues/2229 is a good starting point. It is _probably_ just a matter of finding the right change to Bitcoin Core (https://github.com/bitcoin/bitcoin) and [cherry picking](https://git-scm.com/docs/git-cherry-pick) it to a new branch based on 1.21-dev, then raising a full request from it.

**Comment by aminbadshah** [comment]
Hi, 

So I tried unsuccessfully to compile this last week or so after work. I had a few issues it wouldn't compile, i have some C++ experience. I really want to contribute somehow, i just finished watching the 3 hour podcast by Vitalik. It was really impressive, and he raised a lot of good points with integrations with Doge.

So this is my environment: Windows 10, installed WSL as per the help file.. Got Ubuntu Bash installed on my windows machine..

Here is what i did, please let me know if i'm missing something..

In a Linux Ubuntu WSL Terminal: I did

git clone https://github.com/dogecoin/dogecoin.git

Created a folder called Dogecoin so then in Bash typed  cd dogecoin

I did a checkout of the branch 1.21

Then in Bash this is what i have figured out so far to help with the compilation issues..

PATH=$(echo "$PATH" | sed -e 's/:\/mnt.*//g') # strip out problematic Windows %PATH% imported var 
sudo bash -c "echo 0 > /proc/sys/fs/binfmt_misc/status" # Disable WSL support for Win32 applications.

cd depends
make HOST=x86_64-w64-mingw32
cd ..
./autogen.sh # not required when building from tarball
CONFIG_SITE=$PWD/depends/x86_64-w64-mingw32/share/config.site ./configure --prefix=/
make

I get some compilation errors

![image](https://user-images.githubusercontent.com/11451980/120061858-ff981280-c013-11eb-95f4-a718e18b9daa.png)

I'm I missing a step? any help will be appreciated?

**Reply by rnicoll** [reply]
I have to admit I work on Ubuntu, but hoping we have someone in the community who's working on Windows and can advise.

**Reply by aminbadshah** [reply]
Should i just install Ubuntu as a Dual boot would that make it easier?

**Reply by rnicoll** [reply]
I use Ubuntu 20.04 on a VM, probably simplest to do that or dual-boot, yes.

**Reply by aminbadshah** [reply]
Sweet thanks I will try that out..

**Reply by patricklodder** [reply]
@aminbadshah did you see https://github.com/dogecoin/dogecoin/blob/1.14.4-dev/doc/build-windows.md ? That was recently updated by @michilumin and verified to work on WSL.

**Reply by aminbadshah** [reply]
yeah i'm pretty positive that i went through these set of instructions. I can try it again both ways i'll install Ubuntu on a VM and then also try the WSL as well on Friday and will provide an update on either Friday or Sat once i get through it again..

**Comment by micaelmalta** [comment]
@rnicoll https://github.com/dogecoin/dogecoin/pull/2260

**Comment by rnicoll** [comment]
@aminbadshah I've merged @mmicael1 's patch, can you try that now please?

**Reply by aminbadshah** [reply]
Hi i'm sorry did you mean try it on WSL again or just install it fresh on Ubuntu on Virtual Box? my scrum ends this week at work so i was busy trying to complete some work this weekend. I can try again this comin weekend..

**Reply by rnicoll** [reply]
I'd meant WSL, but honestly I'm good with any scenario which gets you able to compile, fastest.

**Reply by aminbadshah** [reply]
Sweet thanks i'll try it both ways wsl and on virtual box, i'll try it on Friday night and i will let you know..

**Comment by aminbadshah** [comment]
Hi Actually did try this... from https://github.com/dogecoin/dogecoin/blob/1.14.4-dev/doc/build-windows.md
yes this has some good info and i got it to compile... yay...

There are a couple of mistakes on the doc: if you are using Ubuntu18.04 it doesn't support libqrencode4 
sudo apt-get install libssl-dev libboost-all-dev qt5-default libprotobuf-dev libqrencode4 libdb++-dev libdb-dev miniupnpc

I think this is for newer versions of Ubuntu.. i may be mistaken i live in the Windows world plz don't judge me....

Had to use libqrencode3 instead for Ubuntu18.04

so

sudo apt-get install libssl-dev libboost-all-dev qt5-default libprotobuf-dev libqrencode3 libdb++-dev libdb-dev miniupnpc
this should work...

apart from that had one issue where it was complaining about virtualization when running the first command in Ubuntu18.04. So i did enable that from the BIOS, however that did not work. what did work was running the Ubuntu Terminal in Admin Mode.. that got rid of the virtualization error... 

but yeah using windows wsl and working got the wallet to work i guess i'm 7 years late to the party... ok now that i have this running where do i start... i'm like really new to Blockchain, i have experience in the centralized world, but i really want to learn and help out if i can..  do we have any simple bug fixes so that i can get familiar with the code? Like where would you suggest i can start... 

I have like experiences all over machine coding, web, mobile, windows apps for like 7 years running now.. so hopefully i can help out.. lets see..

![image](https://user-images.githubusercontent.com/11451980/121804007-fc4a7c80-cbf8-11eb-83df-4dca7111ef82.png)
